File: src/frontend/lib/admin/models/vm/thing_vm.dart
import 'poll_vm.dart';
import 'settlement_proposal_vm.dart';
import 'thing_validation_verifier_lottery_vm.dart';

class ThingVm {
  final String id;
  final String title;
  final ThingValidationVerifierLotteryVm? lottery;
  final PollVm? poll;
  final SettlementProposalVm? settlementProposal;

  ThingVm.fromMap(Map<String, dynamic> map)
      : id = map['id'],
        title = map['title'],
        lottery = map['lottery'] != null ? ThingValidationVerifierLotteryVm.fromMap(map['lottery']) : null,
        poll = map['poll'] != null ? PollVm.fromMap(map['poll']) : null,
        settlementProposal =
            map['settlementProposal'] != null ? SettlementProposalVm.fromMap(map['settlementProposal']) : null;
}

File: src/Orchestrator/src/Application/Admin/Queries/GetContractsStates/VM/ThingVm.cs
namespace Application.Admin.Queries.GetContractsStates.VM;

public class ThingVm
{
    public required Guid Id { get; init; }
    public required string Title { get; init; }
    public ThingValidationVerifierLotteryVm? Lottery { get; init; }
    public PollVm? Poll { get; init; }
    public SettlementProposalVm? SettlementProposal { get; init; }
}
